Paco-Server/src/com/google/sampling/experiential/server/CloudSQLReportsDaoImpl.java

```diff
@@ -0,0 +1,341 @@
+    ResultSetMetaData rsmd = null;
+    CallableStatement cStmt = null;
+    BlobKey blobKey = null;
+    List<String> participantsLst = null;
+    Connection conn = CloudSQLConnectionManager.getInstance().getConnection();
+    CloudSQLReportsDaoImpl daoImpl = new CloudSQLReportsDaoImpl();
+    int ctr = 0;
+    int colCount = 0;
+   
+    Long startTime = System.currentTimeMillis();
+    participantsLst = daoImpl.getParticipants(expId, who);
+    GcsOutputChannel writeChannel = csfw.getCSWriterChannel(jobId, "application/json", "project-private", jobId);
+    PrintWriter writer = new PrintWriter(Channels.newWriter(writeChannel, "UTF8"));
+        
+    try {
+      cStmt = conn.prepareCall("call ExpQuickStatus(?,?)");
+      for(String eachParticipant : participantsLst) {
+        cStmt.setLong(1, expId);
+        cStmt.setString(2, eachParticipant);
+        ResultSet rs = cStmt.executeQuery();
+        if(ctr++ == 0) {
+          rsmd = rs.getMetaData();
+          colCount = rsmd.getColumnCount();
+        }
+        // traverse result set and create response json object
+        rs.beforeFirst();
+        while (rs.next()) {
+          JSONObject eachRecord = new JSONObject();
+          for (int i=0; i<colCount; i++) {
+            String colName = rsmd.getColumnName(i+1);
+            switch (rsmd.getColumnType(i+1)) {
+              case java.sql.Types.BIGINT:
+                eachRecord.put(colName, rs.getBigDecimal(colName));
+                break;
+              case java.sql.Types.INTEGER:
+                eachRecord.put(colName, rs.getInt(colName));
+                break;
+              case java.sql.Types.VARCHAR:
+                eachRecord.put(colName, rs.getString(colName));
+                break;
+              case java.sql.Types.DATE:
+                eachRecord.put(colName, rs.getDate(colName));
+                break;
+              case java.sql.Types.TINYINT:
+                eachRecord.put(colName, rs.getShort(colName));
+                break;
+              default: 
+                eachRecord.put(colName, rs.getObject(colName));
+            }
+          }// for loop
+          //to cloud storage
+          writer.println(eachRecord);
+          writer.flush();
+          writeChannel.waitForOutstandingWrites();
+        }// while loop
+        
+        long diff = startTime - System.currentTimeMillis();
+        log.info(ctr+ "quick status took"+ diff);
+      }
+    } finally {
+      try {
+        if (cStmt != null) {
+          cStmt.close();
+        }
+        if (conn != null) {
+          conn.close();
+        }
+      } catch (SQLException ex1) {
+        log.warning(ErrorMessages.CLOSING_RESOURCE_EXCEPTION.getDescription()+ ex1);
+      }
+      log.info("stoptime"+ System.currentTimeMillis());
+    }
+    writeChannel.close();
+    blobKey = blobstoreService.createGsBlobKey("/gs/" + bucketName + "/" + jobId);
```

Comment:
the blob key code happens in a couple of places. Can you refactor it into a shared method?